src/core/main.qml

```diff
@@ -188,14 +219,18 @@ Window {
                             pageView.currentItem.focus = true
                             checkWordset()
                         }
-            );
+                        );
         }
         else {
             // Register voices-resources for current locale, updates/downloads only if
             // not prohibited by the settings
             if (!DownloadManager.areVoicesRegistered()) {
                 if (DownloadManager.updateResource(
                             DownloadManager.getVoicesResourceForLocale(ApplicationSettings.locale))) {
+                    DownloadManager.downloadFinished.connect(function() {
+                        checkWordset();
+                        DownloadManager.downloadFinished.disconnect(arguments.callee);
+                    });
```

Comment:
If they are in the PR, then you probably did some change?
I tested it and if we don't have the wordset, then the pop up to download them at start up is displayed twice while with the master branch code, it is displayed only once